File: Chapter 03/Python/section_3_3_2_ocp/after/defensive_formation.py
from .play import Play

"""
A Concrete Strategy.
* ARCHITECTURE NOTE: This class encapsulates one specific behavior. 
Because it's a standalone file, it's easy to test and modify without 
impacting any other part of the system.
"""
class DefensiveFormation(Play):
    def execute(self):
        print("  [Action] Getting into defensive position…")

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/section_3_3_2_ocp/after/DefensiveFormation.java
package com.grokkingsoftwarearchitecture.chapter03.section_3_3_2_ocp.after;

/**
 * A concrete implementation of the Play interface.
 */
public class DefensiveFormation implements Play {
    public void execute() {
        System.out.println("  [Action] Getting into defensive position…");
    }
}